File: server/php/src/Configuration/MerchantClientConfiguration/MerchantClientConfiguration.php
<?php

namespace OnlinePayments\ExampleApp\Configuration\MerchantClientConfiguration;

use Dotenv\Dotenv;
use OnlinePayments\ExampleApp\Configuration\Exceptions\MissingCredentialsException;
use OnlinePayments\Sdk\Client as Client;
use OnlinePayments\Sdk\CommunicatorConfiguration;
use OnlinePayments\Sdk\Authentication\V1HmacAuthenticator;
use OnlinePayments\Sdk\Communicator;
use OnlinePayments\Sdk\Merchant\MerchantClient;

class MerchantClientConfiguration
{
    private string $merchantId;
    private string $apiKey;
    private string $apiSecret;
    private string $apiEndpoint;
    private string $integrator;
    private string $allowedOrigin;

    private Client $client;
    private MerchantClient $merchantClient;

    public function __construct(string $rootPath)
    {
        $dotenv = Dotenv::createImmutable($rootPath);
        $dotenv->load();

        $this->merchantId = getenv('MERCHANT_ID') ?: ($_ENV['MERCHANT_ID'] ?? '');
        $this->apiKey     = getenv('API_KEY') ?: ($_ENV['API_KEY'] ?? '');
        $this->apiSecret  = getenv('API_SECRET') ?: ($_ENV['API_SECRET'] ?? '');
        $this->apiEndpoint= getenv('API_ENDPOINT') ?: ($_ENV['API_ENDPOINT'] ?? '');
        $this->integrator = getenv('INTEGRATOR') ?: ($_ENV['INTEGRATOR'] ?? 'INTEGRATOR');
        $this->allowedOrigin = getenv('ALLOWED_ORIGIN') ?: ($_ENV['ALLOWED_ORIGIN'] ?? '');

        if (!$this->merchantId || !$this->apiKey || !$this->apiSecret || !$this->apiEndpoint || !$this->integrator || !$this->allowedOrigin) {
            throw new MissingCredentialsException("Missing required WL SDK credentials in environment.");
        }

        $this->createClient();
        $this->createMerchantClient();
    }

    private function createClient(): void
    {
        $communicatorConfig = new CommunicatorConfiguration(
            $this->apiKey,
            $this->apiSecret,
            $this->apiEndpoint,
            $this->integrator
        );

        $authenticator = new V1HmacAuthenticator($communicatorConfig);
        $communicator  = new Communicator($communicatorConfig, $authenticator);

        $this->client = new Client($communicator);
    }

    private function createMerchantClient(): void
    {
        $this->merchantClient = $this->client->merchant($this->merchantId);
    }

    public function getClient(): Client
    {
        return $this->client;
    }

    public function getMerchantClient(): MerchantClient
    {
        return $this->merchantClient;
    }
}

File: server/dotnet/sdk-dotnet-example/Configuration/MerchantClientConfiguration/MerchantClientConfiguration.cs
using Configuration.Exceptions;
using OnlinePayments.Sdk;
using OnlinePayments.Sdk.Authentication;
using OnlinePayments.Sdk.Communication;
using OnlinePayments.Sdk.Merchant;
using Microsoft.Extensions.Options;
using Microsoft.Extensions.Logging;

namespace Configuration.MerchantClientConfiguration;

public class MerchantClientConfiguration
{
    private readonly ILogger<MerchantClientConfiguration> logger;

    private readonly AppSettings.AppSettings appSettings;
    private readonly IClient client;
    private readonly IMerchantClient merchantClient;

    public MerchantClientConfiguration(
        ILogger<MerchantClientConfiguration> logger,
        IOptions<AppSettings.AppSettings> appSettings
    )
    {
        this.logger = logger;
        this.appSettings = appSettings.Value;
        client = SetupClient(SetupCommunicatorConfiguration());
        merchantClient = SetupMerchantClient(client);
    }

    public IMerchantClient GetMerchantClient()
    {
        return merchantClient;
    }

    private IClient SetupClient(CommunicatorConfiguration communicatorConfiguration)
    {
        try
        {
            if (appSettings.UseCommunicatorConfiguration)
            {
                return CreateClientFromCommunicator(communicatorConfiguration);
            }

            return CreateClient();
        }
        catch (Exception e)
        {
            logger.LogError(e, "An error has occurred during client setup.");
            
            throw new ConfigurationException("An error has occurred during client setup.");
        }
    }

    private IClient CreateClientFromCommunicator(CommunicatorConfiguration communicatorConfiguration)
    {
        IClient newClient = Factory.CreateClient(communicatorConfiguration);
        newClient.CloseIdleConnections(TimeSpan.FromMinutes(10));

        return newClient;
    }

    private IClient CreateClient()
    {
        IConnection connection = new DefaultConnection(TimeSpan.FromSeconds(10000));
        IAuthenticator authenticator =
            new V1HmacAuthenticator(AuthorizationType.V1HMAC, appSettings.ApiKey, appSettings.ApiSecret);
        IMetadataProvider metadataProvider = new MetadataProvider(new MetadataProviderBuilder(appSettings.MerchantId));
        
        Uri apiEndpoint = new Uri(appSettings.ApiEndpoint);
        IClient newClient = Factory.CreateClient(apiEndpoint, connection, authenticator, metadataProvider);
        
        newClient.CloseIdleConnections(TimeSpan.FromMinutes(10));

        return newClient;
    }

    private IMerchantClient SetupMerchantClient(IClient setupClient)
    {
        return setupClient.WithNewMerchant(appSettings.MerchantId);
    }

    private CommunicatorConfiguration SetupCommunicatorConfiguration()
    {
        try
        {
            return new CommunicatorConfiguration
            {
                ApiKeyId = appSettings.ApiKey,
                SecretApiKey = appSettings.ApiSecret,
                ApiEndpoint = new Uri(appSettings.ApiEndpoint),
                Integrator = appSettings.MerchantId,
                AuthorizationType = AuthorizationType.V1HMAC
            };
        }
        catch (Exception ex)
        {
            logger.LogError(ex,
                "An error has occurred during communicator configuration setup. Please check provided credentials.");
            
            throw new ConfigurationException(
                "An error has occurred during communicator configuration setup. Please check provided credentials.");
        }
    }
}